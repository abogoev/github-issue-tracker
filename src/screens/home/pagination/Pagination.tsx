import { StyleSheet, View, ViewProps } from "react-native";
import React, { VFC } from "react";
import PageItem from "./pagenumber/PageItem";

interface Props extends ViewProps {
  numbers: number[];
  activeIndex: number;
  onNext: () => void;
  onPrev: () => void;
  onChangeActiveNumber: (i: number) => void;
}

const Pagination: VFC<Props> = ({
  numbers,
  activeIndex,
  style,
  onNext,
  onPrev,
  onChangeActiveNumber: onChangeActiveIndex,
  ...props
}) => {
  return numbers.length > 0 ? (
    <View style={[styles.container, style]} {...props}>
      {numbers[0] !== 1 && <PageItem text="Prev" onPress={onPrev} />}
      {numbers.map((n, i) => (
        <PageItem
          key={i}
          text={n.toString()}
          isActive={i === activeIndex}
          onPress={() => onChangeActiveIndex(i)}
        />
      ))}
      <PageItem text="Next" onPress={onNext} />
    </View>
  ) : null;
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "center",
  },
});

export default Pagination;
