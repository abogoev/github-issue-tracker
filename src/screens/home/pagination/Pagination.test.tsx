import React from "react";
import "react-native";
import Pagination from "./Pagination";
import { create } from "react-test-renderer";

describe("Pagination", () => {
  it("should render correctly", () => {
    const mockFunction = jest.fn();
    let numbers = [4, 5, 6];

    let component = create(
      <Pagination
        numbers={numbers}
        activeIndex={0}
        onPrev={mockFunction}
        onNext={mockFunction}
        onChangeActiveNumber={mockFunction}
      />
    );

    expect(component.root.findByProps({ isActive: true }).props.text).toBe(
      numbers[0].toString()
    );

    expect(component.toJSON()).toMatchSnapshot("with prev");

    numbers = [1, 2, 3];

    component = create(
      <Pagination
        numbers={numbers}
        activeIndex={0}
        onPrev={mockFunction}
        onNext={mockFunction}
        onChangeActiveNumber={mockFunction}
      />
    );

    expect(component.toJSON()).toMatchSnapshot("without prev");
  });

  it("should call the callbacks correctly", () => {
    const onPrevMock = jest.fn();
    const onNextMock = jest.fn();
    const onChangeActiveNumberMock = jest.fn();
    const numbers = [4, 5, 6];

    const component = create(
      <Pagination
        numbers={numbers}
        activeIndex={0}
        onPrev={onPrevMock}
        onNext={onNextMock}
        onChangeActiveNumber={onChangeActiveNumberMock}
      />
    );

    expect(onPrevMock).not.toHaveBeenCalled();
    expect(onNextMock).not.toHaveBeenCalled();
    expect(onChangeActiveNumberMock).not.toHaveBeenCalled();

    component.root.findByProps({ text: "Prev" }).props.onPress();

    expect(onPrevMock).toHaveBeenCalledTimes(1);

    component.root.findByProps({ text: "Next" }).props.onPress();

    expect(onNextMock).toHaveBeenCalledTimes(1);

    const index = 2;
    component.root
      .findByProps({ text: numbers[index].toString() })
      .props.onPress();

    expect(onChangeActiveNumberMock).toHaveBeenCalledWith(index);
  });
});
