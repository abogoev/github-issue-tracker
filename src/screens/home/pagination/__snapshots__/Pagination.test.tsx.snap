// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render correctly: with prev 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "transparent",
        "borderColor": "rgb(208, 215, 222)",
        "borderRadius": 6,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#24292f",
          "fontFamily": "Helvetica",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 20,
        }
      }
    >
      Prev
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "rgb(9, 105, 218)",
        "borderColor": "transparent",
        "borderRadius": 6,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Helvetica",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 20,
        }
      }
    >
      4
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "transparent",
        "borderColor": "rgb(208, 215, 222)",
        "borderRadius": 6,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#24292f",
          "fontFamily": "Helvetica",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 20,
        }
      }
    >
      5
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "transparent",
        "borderColor": "rgb(208, 215, 222)",
        "borderRadius": 6,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#24292f",
          "fontFamily": "Helvetica",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 20,
        }
      }
    >
      6
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "transparent",
        "borderColor": "rgb(208, 215, 222)",
        "borderRadius": 6,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#24292f",
          "fontFamily": "Helvetica",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 20,
        }
      }
    >
      Next
    </Text>
  </View>
</View>
`;

exports[`Pagination should render correctly: without prev 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "rgb(9, 105, 218)",
        "borderColor": "transparent",
        "borderRadius": 6,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Helvetica",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 20,
        }
      }
    >
      1
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "transparent",
        "borderColor": "rgb(208, 215, 222)",
        "borderRadius": 6,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#24292f",
          "fontFamily": "Helvetica",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 20,
        }
      }
    >
      2
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "transparent",
        "borderColor": "rgb(208, 215, 222)",
        "borderRadius": 6,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#24292f",
          "fontFamily": "Helvetica",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 20,
        }
      }
    >
      3
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "transparent",
        "borderColor": "rgb(208, 215, 222)",
        "borderRadius": 6,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#24292f",
          "fontFamily": "Helvetica",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 20,
        }
      }
    >
      Next
    </Text>
  </View>
</View>
`;
